mantidproject/mantid
# Python: Add method to get IPF path

**Description of work.**
Moves some code out of `LoadInstrument` into a module where anyone can call it to use IPF files.
Each commit is stand-alone and the work breaks down as follows:

- Move loop into IPF loading, so it will check each instrument dir itself rather than being told to
- Move IPF loading into Experiment Info and add tests
- Move various static file handling out of the ever-growing `ExperimentInfo` into their own file
- Add Python Export
- Rename the new module to `FileFinderHelpers` (although I'm up for any other name suggestions)

**To test:**
- Fire up mantid and check this gives a sane result:

```python
import mantid.api
mantid.api.FileFinderHelpers.getIPFPath("GEM")  # Should return the path to `GEM_Parameters.xml`
mantid.api.FileFinderHelpers.getIPFPath("NotThere") # Should return nothing
```

*There is no associated issue.* - Was required for TOML work in #28513 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
